src/os/bluestore/StupidAllocator.cc

```diff
@@ -80,6 +81,9 @@ uint64_t StupidAllocator::_aligned_len(
     skew = alloc_unit - skew;
   if (skew > p.get_len())
     return 0;
+  else if (periodic_discard &&
```

Comment:
IMO this function (according to its name) is absolutely unrelated to this check. Suggest to either rename the function or move out of the func (e.g. create another one).

Reply:
I implemented it w/o consistency.
I agree with you, thank you for your opinion!